### PR TITLE
python3Packages.datefinder: init at 0.7.3

### PR DESCRIPTION
**Pull request by de11n**
## Description of changes

Add Python package `datefinder`: https://pypi.org/project/datefinder.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by natsukium** [comment]
@ofborg build python310Packages.datefinder

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 249552` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.datefinder</li>
    <li>python310Packages.datefinder.dist</li>
    <li>python311Packages.datefinder</li>
    <li>python311Packages.datefinder.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/datefinder/default.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+, fetchFromGitHub
+, pytestCheckHook
+, python-dateutil
+, pytz
+, regex
+}:
+
+buildPythonPackage rec {
+  pname = "datefinder";
+  version = "0.7.3";
```
Comment (natsukium) [inline pkgs/development/python-modules/datefinder/default.nix:12]:
```suggestion
  version = "0.7.3";
  format = "setuptools";
```
Would you please add `format`?

Comment (de11n) [reply pkgs/development/python-modules/datefinder/default.nix:12]: Done!

**Review by natsukium — APPROVED** [review]
LGTM, thanks!

**Pull request merged**